theano/compile/mode.py

```diff
@@ -333,7 +353,7 @@ def requiring(self, *tags):
                                               self.provided_optimizer)
         return self.__class__(linker=link, optimizer=opt.requiring(*tags))
```

Comment:
While there, please update including() excluding() and requiring() to use clone().